### PR TITLE
Conversion and unit tests

### PR DESCRIPTION
**Pull request by HenningTimm**
This PR adds a rework of severity level management. Severity levels, including the level NONE that indicates a file/test without violations are now used in all returns. On the user side, this PR introduces the following changes:

- A minimal conversion test that converts a yml to tsv file.
- Tests are now run and required for PRs.
- Add lint `b003` that checks if all titles within the DatasetField block are unique.

### CODE REVIEW

File: yml2block/__main__.py [hunk 1]
```diff
@@ -127,15 +135,25 @@ def return_violations(lint_violations, warn_ec, verbose):
...
         if max_severity == Level.ERROR:
+            print("Errors detected. File(s) cannot safely be converted to TSV.")
             sys.exit(1)
         elif max_severity == Level.WARNING:
+            print(
+                "Warnings detected. File(s) can probably not be safely converted to TSV."
+            )
             sys.exit(warn_ec)
+        elif max_severity == Level.NONE:
+            print("\nAll Checks passed! 🎉 Safe covnersion is possible.\n\n")
```
Comment (Athemis) [inline yml2block/__main__.py:155]: Small typo:
```suggestion
            print("\nAll Checks passed! 🎉 Safe conversion is possible.\n\n")
```

**Comment by Athemis** [comment]
Conversion of my stack of yml files works just fine with this branch.

**Review by Athemis — APPROVED** [review]
Conversion of my stack of yml files works just fine with this branch.

**Comment by HenningTimm** [comment]
Thank you!

**Pull request merged**